keps/sig-cloud-provider/0013-build-deploy-ccm.md

```diff
@@ -0,0 +1,652 @@
+```package cloudprovider
+
+import (
+  // Cloud providers
+  _ "k8s.io/cloudprovider-gce"
+)
+```
+
+#### Build Targets
+
+We then get to the issue of creating a deployable artifact for each cloud provider. There are several artifacts beyond 
+the CCM which are cloud provider specific. These artifacts include things like the deployment scripts themselves, the 
+contents of the add-on manager and the sidecar needed to get CSI/cloud-specific persistent storage to work. Ideally 
+these would then be packaged with a version of the Kubernetes core components (KAS, KCM, Kubelet, …) which have not 
+been statically linked against the cloud provider libraries. However in the short term the K8s/K8s deployable builds 
+will still need to link these binaries against all of the in-tree plugins. We need a way for the K8s/cloud-provider 
+repo to consume artifacts generated by the K8s/K8s repo. For official releases these artifacts should be published to a 
+package repository. From there the cloud providers can pull the cloud agnostic Kubernetes artifact and decorate it 
+appropriate to their cloud. The K8s/Cloud-Provider should be using some sort of manifest file to determine which 
+official K8s/K8s artifact to pull. This allows for things like repeatability in builds, for hot fix builds and also a 
+mechanism for rolling out changes which span K8s/K8s and K8s/Cloud-Provider. For local cloud releases we can use a 
+build convention. We can expect the K8s/K8s and K8s/Cloud-Provider repos to be checked out under the same GOPATH. 
+The K8s/Cloud-Provider can then have a local build target which looks for the K8s/K8s artifacts to be in the 
+appropriate place under the GOPATH (or at a location pointed to be a K8s home environment variable) This allows for 
+both official builds and for developers to easily work at running changes which span K8s/K8s and K8s/Cloud-Provider.
+
+#### K8s/K8s Releases and K8s/CP Releases
+
+One of the goals for this project is to free cloud providers to generate releases as they want. This implies that 
+K8s/CP releases are largely decoupled from K8s/K8s releases. The cadence of K8s/K8s releases should not change. We are 
+assuming that K8s/CP releases will be on a similar or faster cadence as needed. It is desirable for the community for 
+all cloud providers to support recent releases. As such it would be good for them to minimize the lag between K8s/K8s 
+releases and K8s/CP releases. A K8s/CP cannot however release a Kubernetes version prior to that version having been 
+released by K8s/K8s. (So for example the cloud provider cannot make a release with a 1.13 Kubernetes core, prior to 
+K8s/K8s having released 1.13) The ownership and responsibility of publishing K8s/K8s will not change with this project. 
+The K8s/CP releases must necessarily move to being the responsibility of that cloud provider or a set of owners 
+delegated by that cloud provider.
+
+#### Migrating to a CCM Build [WIP]
```

Comment:
IMO I think we should try to reduce the scope of this by removing all the WIPs.

Reply:
Lets chat. Willing to remove the WIP flag from things like [Migrating to a CCM Build] and make it part of the proposal. I think for things like KubeAdm and TestGrid we need a proper answer in the KEP. I don't want to hold up the initial work while we get those answers but we need to track that we need those answers.